Solutions/Corvus.Retry/Corvus/Retry/RetryTaskFactory.cs

```diff
@@ -778,7 +773,7 @@ internal static Task HandleRetry(Task task, Func<Task> createTask, IRetryStrateg
 
                 if (delay != TimeSpan.Zero)
                 {
-                    SleepService?.Sleep(delay);
+                    SleepService.Instance.Sleep(delay);
```

Comment:
Should we not be using the `SleepAsync()` method here and chaining the two tasks together, rather than blocking the thread at this point?

Reply:
I've made a work item to review this as it looks as if this whole bit of code might need a rewrite:

https://github.com/corvus-dotnet/Corvus.Retry/issues/63